qrcode_detection_package/main.py

```diff
@@ -1,56 +1,358 @@
+        It checks if the target ID of the control message matches the QR code scanner node.
+        If it does, the function logs a message indicating the receipt of the control message.
+        Depending on the 'active' flag in the control message, the function either activates
+        or deactivates the QR code scanner node by calling the respective private methods.
+
+        Args: 
+            control_msg: The received control message.
+
+        Returns: 
+            None
+        """
+        if (control_msg.target_id == "qr_code_scanner_node"):
+            self.get_logger().info("Recieved control message")
+            if (control_msg.active):
+                self._activate_()
+            else:
+                self._deactivate_()
+
+    def set_state(self, state):
+        """
+        Sets the state of the node to the new state provided.
+
+        Args: 
+            state: The new state to set the node to.
+
+        Returns: 
+            None
+        """
+        self.node_state = state
 
     def __capture_image(self):
-        # For testing purposes the image is not taken from the camera but loaded a test file
-        # Path to the image file
-        image_path = 'src/qrcode_detection_package/test_image/test.jpg'
-        # Load the image using OpenCV
-        captured_image = cv2.imread(image_path)
+        """
+        Capture an image either from the camera or a test image, depending on the configuration.
+
+        This method captures an image either from the camera or from a test image,
+        depending on the configuration specified by 'config_detection_method'.
+        - If 'config_detection_method' is set to 0, a test image is loaded from a predefined path.
+        - If 'config_detection_method' is set to 1, an image is captured from the camera.
+        - If 'config_detection_method' is set to 2, an image is captured using the libcamera command line tool.
+        The captured image is returned as an OpenCV MatLike object.
+
+        Returns: 
+            OpenCV MatLike representing the captured image.
+        """
+        import cv2
+
+        # Check which detection style should be used
+        if (self.config_detection_method == 0):
+            # Path to the test image
+            test_image_path = 'src/qrcode_detection_package/test_image/test3.png'
+
+            # Load the test image
+            captured_image = cv2.imread(test_image_path)
+        elif (self.config_detection_method == 1):
+            # Initialize a VideoCapture object for the camera
+            cap = cv2.VideoCapture(0)
+
+            # To-Do: Evaluate influence of different camera settings to quality and performance on the real hardware
+            cap.set(cv2.CAP_PROP_FRAME_WIDTH, 4056)
+            cap.set(cv2.CAP_PROP_FRAME_HEIGHT, 3040)
+
+            # Check if the VideoCapture object was opened successfully
+            if not cap.isOpened():
+                self.get_logger().info("Error: Unable to open camera")
```

Comment:
Do you want to just output a logging message or also abort here?

Reply:
you´re right, i will now return None in case of an error (also in the error case below). The recieving function checks again if the image = None anyways